### PR TITLE
Feature/art spring upgrade

### PR DESCRIPTION
**Pull request by mtrupkin**

### CODE REVIEW

File: src/main/java/org/opentestsystem/delivery/logging/RabbitConfiguration.java [hunk 1]
```diff
@@ -84,6 +85,7 @@ public Queue springCloudBusQueue() {
   }
 
   @Bean
+  @Primary
```
Comment (millerts) [inline src/main/java/org/opentestsystem/delivery/logging/RabbitConfiguration.java]: Why is it necessary that primary annotations be added to a shared library?

Comment (mtrupkin) [reply src/main/java/org/opentestsystem/delivery/logging/RabbitConfiguration.java]: Verified that primary annotation is not required.

File: src/main/java/org/opentestsystem/delivery/logging/RabbitConfiguration.java [hunk 2]
```diff
@@ -30,6 +30,7 @@
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
```
Comment (millerts) [inline src/main/java/org/opentestsystem/delivery/logging/RabbitConfiguration.java]: Is this import used?

File: src/main/java/org/opentestsystem/shared/search/persistence/SearchableRepositoryImpl.java [hunk 3]
```diff
@@ -14,15 +14,19 @@
...
 import org.opentestsystem.shared.search.domain.AbstractSearchRequest;
 import org.opentestsystem.shared.search.domain.SearchResponse;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.data.mongodb.core.MongoOperations;
 import org.springframework.data.mongodb.core.query.Query;
 import org.springframework.data.mongodb.repository.query.MongoEntityInformation;
 import org.springframework.data.mongodb.repository.support.SimpleMongoRepository;
+import org.springframework.stereotype.Component;
 
+@Component
```
Comment (millerts) [inline src/main/java/org/opentestsystem/shared/search/persistence/SearchableRepositoryImpl.java]: Why did the Spring upgrade require the addition of this annotation?  Is there an underlying change that I should look at to see why this was necessary?

File: src/main/java/org/opentestsystem/shared/search/persistence/SearchableRepositoryImpl.java [hunk 4]
```diff
@@ -14,15 +14,19 @@
...
 public class SearchableRepositoryImpl<T, ID extends Serializable> extends SimpleMongoRepository<T, ID> implements SearchableRepository<T, ID> {
     private final MongoEntityInformation<T, ID> entityInformation;
     private final MongoOperations mongoOperations;
 
+    @Autowired
```
Comment (millerts) [inline src/main/java/org/opentestsystem/shared/search/persistence/SearchableRepositoryImpl.java]: Same comment as above.  This was working without annotations so I'm curious why we're adding them now.  Or did we remove the XML config?

File: src/main/java/org/opentestsystem/shared/search/persistence/SearchableRepositoryImpl.java [hunk 5]
```diff
@@ -23,6 +23,7 @@
     private final MongoEntityInformation<T, ID> entityInformation;
     private final MongoOperations mongoOperations;
 
+
```
Comment (millerts) [inline src/main/java/org/opentestsystem/shared/search/persistence/SearchableRepositoryImpl.java]: Extra space.

File: src/main/java/org/opentestsystem/shared/search/persistence/SearchableRepository.java [hunk 6]
```diff
@@ -9,12 +9,32 @@
...
 
 import org.opentestsystem.shared.search.domain.AbstractSearchRequest;
 import org.opentestsystem.shared.search.domain.SearchResponse;
+import org.springframework.data.domain.Sort;
 import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.data.repository.PagingAndSortingRepository;
 
 
-public interface SearchableRepository<T, ID extends Serializable> extends MongoRepository<T, ID> {
+public interface SearchableRepository<T, ID extends Serializable> extends PagingAndSortingRepository<T, ID> {
```
Comment (millerts) [inline src/main/java/org/opentestsystem/shared/search/persistence/SearchableRepository.java:21]: Did we somehow lose this change?  I'm surprised this needed to be modified by just bumping the version of shared build.

**Review by millerts — APPROVED** [review]

File: src/main/java/org/opentestsystem/shared/search/persistence/SearchableRepository.java (hunk 6 above)
Comment (millerts) [reply src/main/java/org/opentestsystem/shared/search/persistence/SearchableRepository.java:21]: This changed due to mongo upgrades

**Pull request merged**